services/bundle_analysis/comparison.py

```diff
@@ -20,23 +20,31 @@
 
 
 class ComparisonLoader:
-    def __init__(self, pull: EnrichedPull):
-        self.pull = pull
+    def __init__(self, base_commit: Commit | None, head_commit: Commit | None):
+        self._base_commit: Commit | None = base_commit
+        self._head_commit: Commit | None = head_commit
+
+    @classmethod
+    def from_EnrichedPull(cls, pull: EnrichedPull) -> "ComparisonLoader":
```

Comment:
```suggestion
    def from_EnrichedPull(cls, pull: EnrichedPull) -> ComparisonLoader:
```

I believe this is now valid syntax in Python 3.12

Reply:
My IDE complains that it is not defined. And the container fails with
```
NameError: name 'ComparisonLoader' is not defined
```

So maybe not? I might be missing something

Reply:
Also found out that `from typing import Self` exists, and that it seems to be even a better fit when inheriting from base classes, as it will just forward the concrete type that methods are being called on. And its less to type (apart from the import)